### PR TITLE
ToggleHDR: Check 60 Hz mode for HDR support if needed

### LINKED ISSUE #9: Switch to 60 Hz when HDR isn't detected?
I have a 4K HDR display that supports up to 144 Hz (in both SDR and HDR). I have it set to "look like" 1440p (with HiDPI), so it behaves like a 5K display. Unfortunately, recent versions of macOS don't support HDR in "looks like 5K" mode even though the display itself supports it. I was able to create a tweaked version of ToggleHDR that does the following:

If HDR is not supported by the display, then:

1. Record the original mode (and write the mode number and `display.name` to `UserDefaults`). 
2. Look for a 60 Hz version of the same mode.
3. If it finds it, switch to that mode.
4. If the display then supports HDR, toggle HDR on.
5. If the display _still_ doesn't support HDR, restore the original mode and print an error.

The reason for storing the data to `UserDefaults` in step 1 is that if "toggle" is requested and it's already in HDR mode, and the `display.name` matches the one stored in UserDefaults, and the mode number successfully creates a mode, it will then restore that mode (so that switching HDR off doesn't leave you stuck in 60 Hz).

Even though I got all this working, my code isn't very clean, so I'm wondering if you are interested in implementing something similar, but doing it right. Or, alternatively, if you are interested in me cleaning it up and submitting a PR. (I probably won't bother to clean it up if you're not interested in accepting a PR.) My code does one other thing, which is if the command line doesn't contain any arguments at all (other than the executable), and the system only contains 1 display, it will assume that the user wants to toggle HDR on the sole display.

### PR DESCRIPTION
**Pull request by tcobbs**
If the current display mode does not support HDR on a specified or the only display, check to see if a 60 Hz mode on the same display supports HDR before failing.

Also, if there are no arguments on the command line and there is only one display, toggle HDR on that display.

Finally, store the original SDR display mode in UserDefaults so that when the user switches back they won't be stuck in 60 Hertz.

Fixes #9

### CODE REVIEW

File: ToggleHDR.swift [hunk 1]
```diff
@@ -58,105 +50,121 @@ func bool(_ arg: String) -> Bool? {
...
     }
 
     defer {
         print("")
         printDisplays(displays)
     }
-    let arg = CommandLine.arguments[1].lowercased()
+    let try60Hz = CommandLine.arguments.contains("--try-60-hz")
+    let args = CommandLine.arguments.filter { arg in !["--try-60-hz"].contains(arg) }
+    let arg = args[1].lowercased()
```
Comment (tcobbs-bentley) [inline ToggleHDR.swift:141]: Won't this crash if `--try-60-hz` is the only command line argument?

File: ToggleHDR.swift [hunk 2]
```diff
@@ -58,105 +50,121 @@ func bool(_ arg: String) -> Bool? {
...
         return
     }
 
     defer {
         print("")
         printDisplays(displays)
     }
-    let arg = CommandLine.arguments[1].lowercased()
+    let try60Hz = CommandLine.arguments.contains("--try-60-hz")
+    let args = CommandLine.arguments.filter { arg in !["--try-60-hz"].contains(arg) }
```
Comment (tcobbs-bentley) [inline ToggleHDR.swift:140]: Isn't this equivalent to the easier to read (and faster?) `arg != "--try-60-hz"`?

File: ToggleHDR.swift (hunk 1 above)
Comment (tcobbs) [reply ToggleHDR.swift:141]: Sorry for using my work account for these two comments. I can confirm here that they both come from the original PR's submitter (me).

Comment (alin23) [reply ToggleHDR.swift:141]: Indeed, I'll fix it. Thanks for catching it!

File: ToggleHDR.swift (hunk 2 above)
Comment (alin23) [reply ToggleHDR.swift:140]: That argument can be in any position, not just the first one. So no they are not equivalent.

**Pull request merged**